src/ray/gcs/gcs_server/gcs_placement_group_manager.h

```diff
@@ -284,9 +292,11 @@ class GcsPlacementGroupManager : public rpc::PlacementGroupInfoHandler {
   /// execute_after).
   boost::asio::io_context &io_context_;
 
-  /// Callback of placement_group registration requests that are not yet flushed.
-  absl::flat_hash_map<PlacementGroupID, StatusCallback>
-      placement_group_to_register_callback_;
+  /// Callbacks of pending `RegisterPlacementGroup` requests.
+  /// Maps placement group ID to placement group registration callbacks, which is used to
+  /// filter duplicated messages from a driver/worker caused by some network problems.
+  absl::flat_hash_map<PlacementGroupID, std::vector<StatusCallback>>
```

Comment:
Why change it to `placement_group_to_register_callbacks_`?

Reply:
just like the registration of `Actor`, there will be the deconstruction problem if we just ignore the previous callback, I've checked this from @raulchen

Reply:
@clay4444 Can you describe the specific problem in detail? thx

Reply:
hm... I'm not sure whether I can explain this clearly, the general reason is that the `GrpcClient` will not be deconstructed since we just ignore the callback and no response was sent to the client

Reply:
If it is a network exception, reply also cannot be returned.